src/utlist.h

```diff
@@ -349,6 +349,37 @@ do {
   }                                                                                            \
 } while (0)
 
+#define LL_INSERT_INORDER(head,add,cmp)                                                        \
+    LL_INSERT_INORDER2(head,add,cmp,next)
+
+#define LL_INSERT_INORDER2(head,add,cmp,next)                                                  \
+do {                                                                                           \
+  LDECLTYPE(head) _tmp;                                                                        \
```

Comment:
FYI: in my version of this patch I make the `NO_DECLTYPE` codepath work as well.
(You can test it with `make pedantic EXTRA_CFLAGS=-DNO_DECLTYPE`.)